Source/Scene/Material.js

```diff
@@ -690,6 +690,16 @@ define([
         checkForValidProperties(uniforms, materialNames, duplicateNameError, false);
     }
 
+    function isMaterialFused(shaderComponent, material) {
+        for ( var subMaterialId in material._template.materials) {
+            if (shaderComponent.indexOf(subMaterialId) > -1) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
Use `hasOwnProperty` - eslint doesn't catch it but it's good practice.

```js
    function isMaterialFused(shaderComponent, material) {
        var materials = material._template.materials;
        for (var subMaterialId in materials) {
            if (materials.hasOwnProperty(subMaterialId)) {
                if (shaderComponent.indexOf(subMaterialId) > -1) {
                    return true;
                }
            }
        }

        return false;
    }
```

Reply:
If it doesn't catch it, we should fix that.  https://eslint.org/docs/rules/guard-for-in is a candidate.  Can you write up an issue so we look at it?

Reply:
Opened https://github.com/AnalyticalGraphicsInc/cesium/issues/8100.